Added complete and cancel buttons to PickupEvent Edit Page

WishingWell13 opened the pull request:
Also fixed text wrap issue

Also updated Button component to take in class names

<!-- Fill in all spots surrounded by brackets and confirm all check boxes after confirming completion of the tasks -->

# Info

Closes #203.

<!-- If there is no issue for this pull request yet, please create one or
delete this line if the pull request is for a very minor tweak. -->


 Implements the `complete`/`fulfill` and `cancel` functionality on the frontend for Pickup Events.

<!-- What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context. -->



## Changes

- Added complete and cancel buttons to Pickup Event Edit Page and Home Page
- Also fixed text wrap issue on admin event page
- Also updated Button component to take in class names


# Type of Change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) <!-- Edit this type of change if you select this -->

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [x] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

<img width="441" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/543b14bf-8117-4459-a243-0aacd045f726">

<img width="504" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/5e190637-9c48-4c7a-915b-f2455e93ce1d">

<img width="534" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/0a62b600-9e47-43fb-a882-647cfac3bdf3">


<!-- If you made any visual changes to the website, please include relevant screenshots below. -->

WishingWell13 committed:
Added complete and cancel buttons to PickupEvent Edit Page

Also fixed text wrap issue

Also updated Button component to take in class names

WishingWell13 committed:
Added buttons to admin pickup home page

WishingWell13 committed:
Merge branch 'main' into andy/completeCancelPickupEvents

SheepTester commented on src/components/admin/event/AdminPickupEvent/AdminPickupEventForm.tsx:
```diff
@@ -245,13 +277,38 @@ const AdminPickupEventForm = ({ mode, defaultData = {}, token, upcomingEvents }:
+              className={style.expandButton}
+            >
               Discard changes
             </Button>
```
I think you should swap the complete and discard buttons so the buttons are grouped by what they do. I think grouping it that way might be better than keeping the blues and reds separate, as it currently is

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/15746db7-b911-4cf3-8373-ebe61e3cc04e)

SheepTester commented on src/components/admin/event/AdminPickupEvent/style.module.scss:
```diff
@@ -32,6 +32,12 @@
 .submitButtons {
```
You should add `flex-wrap: wrap;` here for mobile

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/55248c2d-f4c8-4466-8c39-db6dd732b031)

SheepTester commented on src/components/admin/event/AdminPickupEvent/style.module.scss:
```diff
@@ -32,6 +32,12 @@
+  .expandButton {
+    height: auto;
+    min-height: 2.5rem;
+  }
```
I think we should enable this for every button (i.e. move this to `Button` so it applies to all buttons) since there's no reason to prevent them from expanding

SheepTester commented on src/components/admin/store/PickupEventCard/index.tsx:
```diff
@@ -1,8 +1,13 @@
 import { PublicOrderPickupEvent } from '@/lib/types/apiResponses';
 import { OrderPickupEventStatus } from '@/lib/types/enums';
 import { formatEventDate } from '@/lib/utils';
+import { Button } from '@mui/material';
```
I think you imported the wrong `Button`

SheepTester commented on src/components/admin/store/PickupEventCard/index.tsx:
```diff
@@ -45,6 +51,22 @@ const PickupEventCard = ({ pickupEvent }: PickupEventCardProps) => {
+        }}
+      >
+        <Typography variant="h5/bold">Cancel Pickup Event</Typography>
+      </Button>
```
I think you should move these buttons to the pickup event page, next to "Edit Pickup Event"

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/1ccd8f97-7a85-42f0-bdd2-90152b923c00)

because it's probably a bad idea to have buttons inside links (clicking on the button also clicks the link)

SheepTester commented on src/lib/api/EventAPI.ts:
```diff
@@ -177,6 +179,44 @@ export const deletePickupEvent = async (token: string, pickupEvent: UUID): Promi
+export const cancelPickupEvent = async (
+  token: string,
+  uuid: UUID
+): Promise<CancelOrderPickupEventResponse> => {
```
since it looks like `CancelOrderPickupEventResponse` doesn't return anything meaningful, you can just return `Promise<void>` instead

SheepTester reviewed (approved):
I think you should hide the complete/cancel buttons if the event isn't active

nitpicks aside, this looks good! 🦀🦀

WishingWell13 committed:
Moved complete and edit buttons to single pickup event page

Updated button styling for pickup event page

Updated common Button component styling to auto expand to content

Hide the complete/cancel buttons if the event isn't active

WishingWell13 committed:
Merge branch 'main' into andy/completeCancelPickupEvents

SheepTester commented on src/components/common/Button/style.module.scss:
```diff
@@ -8,19 +8,22 @@
   &[data-size='small'] {
-    height: 2rem;
+    height: auto;
+    min-height: 2rem;
```
you don't have to set `height: auto` again since it's already set to auto above

SheepTester reviewed (approved):
yeah these changes look good

WishingWell13 committed:
Removed unnecessary CSS

WishingWell13 committed:
Merge branch 'andy/completeCancelPickupEvents' of https://github.com/acmucsd/membership-portal-ui-v2 into andy/completeCancelPickupEvents
